### PR TITLE
Allow enabling end-to-end encryption again

### PR DESCRIPTION
**Pull request by j-f1**
I would greatly appreciate feedback on this wording:

> Encryption is not enabled. Your data is still stored on your server, but if you do not fully trust the security of the server, you can enable encryption to ensure that your data is encrypted on the server.

### CODE REVIEW

**Comment by j-f1** [comment]
It might also be nice to either support disabling encryption or clarify that enabling it is a one-way operation.

**Comment by rich-howell** [comment]
@j-f1 I think we should add this to the documentation ....

**Comment by MatissJanis** [comment]
👋 Heads up: I think e2e encryption is definitely something we want to bring back. But adding it is a high-risk operation, so we need to be extra-sure it works as expected.

We'll get the responsive read-only app support out the door in the next release. After that I'll do a deeper review & testing of this PR and we can get it in the release afterwards. So please hang in a bit longer!

File: packages/desktop-client/src/components/settings/Encryption.js [hunk 1]
```diff
@@ -19,30 +22,41 @@ export default function EncryptionSettings({ prefs, pushModal }) {
...
-        <strong>End-to-end encryption</strong> is not enabled. Any data on our
-        servers is still stored safely and securely, but it's not end-to-end
-        encrypted which means we have the ability to read it (but we won't). If
-        you want, you can use a password to encrypt your data on our servers.
+        <strong>End-to-end encryption</strong> is not enabled. Any data on the
+        server is still protected by the server password, but it's not
+        end-to-end encrypted which means the server owners have the ability to
+        read it. If you want, you can use an additional password to encrypt your
+        data on the server.{' '}
+        <a href="https://actualbudget.github.io/docs/Getting-Started/sync/#encryption">
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/settings/Encryption.js:46]: **💬 suggestion:** ‏
```suggestion
        <a href="https://actualbudget.github.io/docs/Getting-Started/sync/#encryption" target="_blank">
```

File: packages/desktop-client/src/components/settings/Encryption.js [hunk 2]
```diff
@@ -19,30 +22,41 @@ export default function EncryptionSettings({ prefs, pushModal }) {
...
+        </a>
+      </Text>
+    </ButtonSetting>
+  ) : (
+    <ButtonSetting button={<Button disabled>Enable encryption…</Button>}>
+      <Text>
+        <strong>End-to-end encryption</strong> is not available when running
+        without a server. Budget files are always kept unencrypted locally, and
+        encryption is only applied when sending data to a server.{' '}
+        <a href="https://actualbudget.github.io/docs/Getting-Started/sync/#encryption">
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/settings/Encryption.js:57]: **💬 suggestion:** ‏
```suggestion
        <a href="https://actualbudget.github.io/docs/Getting-Started/sync/#encryption" target="_blank">
```

File: packages/desktop-client/src/components/modals/CreateEncryptionKey.js [hunk 3]
```diff
@@ -91,6 +91,11 @@ export default function CreateEncryptionKey({
                     will need to reset them. Actual will automatically take you
                     through this process.
                   </li>
+                  <li>
+                    If you reuse the password you log in to the server here,
+                    anybody else who has the server password will be able to
+                    decrypt your data.
+                  </li>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/CreateEncryptionKey.js:98]:
```suggestion
                  <li>
                    It is recommended for the encryption password to be different than the log-in password in order to better protect your data.
                  </li>
```

**Review by MatissJanis — COMMENTED** [review]
Code looks good for the most part. Also did some manual tests and everything seems to be working OK.

Why was this not enabled earlier @jlongster ? This seems like such a low-hanging fruit. I must be missing some crucial piece of information why this was initially disabled.

File: packages/desktop-client/src/components/modals/CreateEncryptionKey.js (hunk 3 above)
Comment (j-f1) [reply packages/desktop-client/src/components/modals/CreateEncryptionKey.js:98]: Ah right, theoretically a malicious server could log your password and use it to decrypt the file. So reuse would throw out all the security guarantees e2e offers 👍

**Comment by jlongster** [comment]
> Why was this not enabled earlier @jlongster ?

Because originally it wasn't compatible with the way the server worked. I attempted to take a different approach for the server when open-sourcing, which was to run the fully app on the server. The server basically became another client the synced changes normally. Was thinking that would make it easy to expose HTTP APIs etc if you wanted.

It caused problems though because there are things that need to be changed for that to work (various things like schedules shouldn't running on the server etc).

In that method the app was running against local files, just like the electron app. It read a `sqlite` file locally from disk. e2e encryption requires the files to be blobs, and stored encrypted such that the server can't access them at all.

Since we switched backed to the "simple" method of syncing, which stores the files as blobs and messages in a simple database and never has the read the data, we can turn this back on!

**Comment by MatissJanis** [comment]
I'm happy to merge this after the lint issues are fixed @j-f1

**Comment by MatissJanis** [comment]
Actually: cc @rich-howell 

Should we update docs before this is merged?

**Comment by j-f1** [comment]
https://github.com/actualbudget/docs/pull/69

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**